src/golang/lib/workflow/dag/workflow_dag.go

```diff
@@ -64,29 +64,20 @@ type workflowDagImpl struct {
 // by the artifact's original ID.
 func computeArtifactSignatures(
 	dbOperators map[uuid.UUID]db_operator.DBOperator,
-	dbArtifacts map[uuid.UUID]db_artifact.DBArtifact,
+	opIDsByInputArtifact map[uuid.UUID][]uuid.UUID,
```

Comment:
Curious what's the motivation here?

Reply:
Shoot I'm having a hard time recalling the exact motivation, since this was done a while back. I think I wanted to move the construction of all the dag maps up into the constructor, instead of the helpers, in an effort to centralize all the data structures we needed in one place, so that future refactors would be easier.

Reply:
lol sg, as long as it works let's worry about that later. Two questions just for this change -
* does `numArtifacts` the same as the size of map `opIDsByInputArtifact`?
* Does this change only modify the interface but not the behavior in terms of how things get hashed?

Reply:
That map is expected to have <= numArtifacts. Terminal artifacts are not included, since they do not have any operators operating on them. The reason this works is because it is legal to loop over `opIDsByInputArtifact[terminalArtifactID]` with range, even if no entry exists, as it will simply not enter the for loop.

This does not change the behavior of how things get hashed at all, the signature of an artifact is still `hash(input artifact signatures +  Optional[parameter value string] + output artifact id])`, it's just been extended to handle multiple output artifact ids.